src/main/index.js

```diff
@@ -597,6 +607,82 @@ function runApp() {
     }
   })
 
+  function createTray(window) {
+    function click(window, close = false) {
+      if (!close) {
+        window.show()
+      } else if (tray.windows.length) {
+        window.destroy()
+      }
+
+      if (BrowserWindow.getAllWindows().length === 0) {
+        handleQuit()
+      }
+
+      tray.windows.splice(tray.windows.findIndex(item => item.id === window.id), 1)
+
+      if (tray.windows.length) {
+        createContextMenu()
+      } else {
+        tray.destroy()
+        tray = null
+      }
+    }
+
+    function createContextMenu() {
+      const menuItems = []
+      tray.windows.forEach(window => {
+        menuItems.push({
+          label: window.title,
+          submenu: [
+            {
+              label: 'Show',
+              click: () => click(window)
+            },
+            {
+              label: 'Close',
+              click: () => click(window, true)
+            }
+          ]
+        })
+      })
+
+      menuItems.push(
+        {
+          type: 'separator'
+        },
+        {
+          label: 'Quit',
+          click: handleQuit
+        }
+      )
+
+      const menu = Menu.buildFromTemplate(menuItems)
+      tray.setContextMenu(menu)
+    }
+
+    if (tray) {
+      tray.windows.push(window)
+      createContextMenu()
+      return
+    }
+
+    const icon = process.env.NODE_ENV === 'development'
+      ? readFileSync(path.join(__dirname, '..', '..', '_icons', 'iconColor.png'))
+      : readFileSync(path.join(path.dirname(__dirname), '_icons', 'iconColor.png'))
+
+    tray = new Tray(nativeImage.createFromBuffer(icon))
+    tray.windows = [window]
```

Comment:
Please just use a normal variable for the windows, instead of unnecessarily adding a new property to the tray object. Apart from being messy (just because JavaScript lets you modify built-ins doesn't mean it is actually a good idea), it will also cause problems once we switch to TypeScript.